sdlbook.c

```diff
@@ -729,27 +752,37 @@ int main(int argc, char **argv) {
 							goto dun_goofed;
 						case SDLK_KP_PLUS:
 							need_redraw = change_scale(+10);
+							change_scroll_h(+96);
```

Comment:
why do you start scrolling when the scale is changed ?

Reply:
if the initial X position is modified and you try to scale up, then after reaching `ezsdl_get_width() > (int)page_dims.w` the scaling doesn't adjust the screen and you're left with free space on right border which gets blitted with random stuff. In general you want to make use of whole screen and don't look at blank part, so I'm re-centering - more of a remedy to a symptom 😅

Reply:
well, i guess leave that remedy away for the moment. it's unrelated to the actual task at hand, which is to add horizontal scrolling. then we gotta find a way to tell the engine that the borders need to be redrawn in certain situations. i've got one pdf here that has a few pages with horizontal illustrations, probably those were photographed with both pages visible, that have different (bigger) horizontal res. and the behaviour there is also everything but ideal; especially that the borders aren't redrawn when we get back to smaller pages.

Reply:
ah didn't though there could be different page sizes...yup not ideal then